lib/BackbeatConsumer.js

```diff
@@ -435,6 +449,13 @@ class BackbeatConsumer extends EventEmitter {
                         entry.timestamp / 1000);
                     return undefined;
                 });
+                // Immediately try to consume more messages
+                // if tasks have completed since the last consume
+                if (this._tasksCompletedSinceLastConsume) {
```

Comment:
should this not be merged with the next block (line 460) : i.e. just add a `_tasksCompletedSinceLastConsume` condition to go there? (and cleanup `_tasksCompletedSinceLastConsume` anyway, it does no harm)

Reply:
ok, this is done separately to avoid the extra 1 sec delay?

Reply:
Yes, we should consume immediately after with no delay. That last block is run when Kafka doesn't have enough messages to consume or when it returns an error, i also didn't want to merge the logics.